### PR TITLE
src/lib/server/cf_util.c: Check if 'CONF_ITEM' is valid

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: src/lib/server/cf_util.c [hunk 1]
```diff
@@ -527,6 +531,8 @@ char const *_cf_filename(CONF_ITEM const *ci)
  */
 bool cf_item_is_section(CONF_ITEM const *ci)
 {
+	if (!ci) return NULL;
```
Comment (alandekok) [inline src/lib/server/cf_util.c]: return `false`

File: src/lib/server/cf_util.c [hunk 2]
```diff
@@ -539,6 +545,8 @@ bool cf_item_is_section(CONF_ITEM const *ci)
  */
 bool cf_item_is_pair(CONF_ITEM const *ci)
 {
+	if (!ci) return NULL;
```
Comment (alandekok) [inline src/lib/server/cf_util.c]: return `false`

File: src/lib/server/cf_util.c [hunk 3]
```diff
@@ -551,6 +559,8 @@ bool cf_item_is_pair(CONF_ITEM const *ci)
  */
 bool cf_item_is_data(CONF_ITEM const *ci)
 {
+	if (!ci) return NULL;
```
Comment (alandekok) [inline src/lib/server/cf_util.c]: return `false`

Comment (jpereira) [reply src/lib/server/cf_util.c]: @alandekok done. My `ctrl-c/ctrl-v` ....  <sigh>

**Comment by alandekok** [comment]
Thinking about this, it may be worth using `CC_HINT(nonnull)` instead.  The difficulty is that the compilers have a lot of false positives when they complain about it.

**Comment by jpereira** [comment]
We have several cases that are walking through a linked-list. so, the compiler never will catch it using `CC_HINT(nonnull)`. (e.g I faced before and fixed it in  https://github.com/FreeRADIUS/freeradius-server/commit/2069ef00b285d5bd408b3caaacefa46449567e40)

**Pull request merged**